File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/viewmodel/javacode/BaseLifecycleActivity.java
package com.ponthaitay.lifecycleactivity.googleio2017.viewmodel.javacode;

import android.arch.lifecycle.LifecycleRegistry;
import android.arch.lifecycle.LifecycleRegistryOwner;
import android.support.v7.app.AppCompatActivity;

public class BaseLifecycleActivity extends AppCompatActivity implements LifecycleRegistryOwner {

    private final LifecycleRegistry mRegister = new LifecycleRegistry(this);

    public BaseLifecycleActivity() {

    }

    @Override
    public LifecycleRegistry getLifecycle() {
        return mRegister;
    }
}

File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/viewmodel/kotlincode/BaseLifecycleActivity.kt
package com.ponthaitay.lifecycleactivity.googleio2017.viewmodel.kotlincode

import android.arch.lifecycle.LifecycleRegistry
import android.arch.lifecycle.LifecycleRegistryOwner
import android.support.v7.app.AppCompatActivity

open class BaseLifecycleActivity : AppCompatActivity(), LifecycleRegistryOwner {

    private val mRegister = LifecycleRegistry(this)

    override fun getLifecycle(): LifecycleRegistry {
        return  mRegister
    }
}
